adyenv6core/src/com/adyen/v6/facades/DefaultAdyenCheckoutFacade.java

```diff
@@ -698,6 +693,16 @@ public void initializeCheckoutData(Model model) {
         modelService.save(cartModel);
     }
 
+    private boolean filterPaymentMethods(PaymentMethod paymentMethod) {
```

Comment:
We were discussing some way to make this method more readable, it has a lot of `!`negation operators that make it hard to understand. We came up with 

    private boolean isHiddenPaymentMethod(PaymentMethod paymentMethod) {
        String paymentMethodType = paymentMethod.getType();

        if (paymentMethodType == null
                || paymentMethodType.isEmpty()
                || paymentMethodType.equals("scheme")
                || paymentMethodType.equals("bcmc")
                || paymentMethodType.equals("bcmc_mobile_QR")
                || paymentMethodType.equals(PAYMENT_METHOD_IDEAL)
                || (paymentMethodType.contains("wechatpay") && !paymentMethodType.equals("wechatpayWeb"))
                || paymentMethodType.startsWith(PAYMENT_METHOD_BOLETO)) {
            return true;
        }

        return false;
    }

and on the caller method change to
`                                                                 .filter(paymentMethod -> ! isHiddenPaymentMethod(paymentMethod))`
What do you think?

Reply:
Looks good. My question is still if we need to have some payment method types as strings and some of them as constants.

Reply:
Yes, I haven't changed that, but I think it would be nice to not mix both. 
The constants are actually imported from a different class `Adyenv6coreConstants`, maybe we can add the others there?

Reply:
yes, we can add more constants. Lets make it uniform.

Reply:
I like the || option given by @martinsrenato. Also, method name isHiddenPaymentMethod is more appropriate and descriptive.